phoenix-core/src/main/java/org/apache/phoenix/coprocessor/ScanRegionObserver.java

```diff
@@ -234,6 +234,12 @@ protected RegionScanner doPostScannerOpen(final ObserverContext<RegionCoprocesso
                     scan.getAttribute(QueryConstants.LAST_SCAN) != null);
         }
         final OrderedResultIterator iterator = deserializeFromScan(scan,innerScanner);
+        
+        if(iterator != null){
+            iterator.setMemoryManager(GlobalCache.getTenantCache(c.getEnvironment(), tenantId).getMemoryManager());
+        }
+
+
```

Comment:
Is it possible to put this MemoryManager into constructor other than using the set method?